iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -457,17 +462,20 @@ public List<ColumnStatisticsObj> getColStatistics(org.apache.hadoop.hive.ql.meta
     Table table = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
     String statsPath = getStatsPath(table).toString();
     LOG.info("Using stats from puffin file at: {}", statsPath);
+    return readColStats(table, statsPath).equals(null) ? null : readColStats(table, statsPath).getStatsObj();
+  }
+
+  private ColumnStatistics readColStats(Table table, String statsPath) {
     try (PuffinReader reader = Puffin.read(table.io().newInputFile(statsPath)).build()) {
       List<BlobMetadata> blobMetadata = reader.fileMetadata().blobs();
-      Map<BlobMetadata, List<ColumnStatistics>> collect =
-          Streams.stream(reader.readAll(blobMetadata)).collect(Collectors.toMap(Pair::first,
-              blobMetadataByteBufferPair -> SerializationUtils.deserialize(
-                  ByteBuffers.toByteArray(blobMetadataByteBufferPair.second()))));
-      return collect.get(blobMetadata.get(0)).get(0).getStatsObj();
+      Map<BlobMetadata, List<ColumnStatistics>> collect = Streams.stream(reader.readAll(blobMetadata)).collect(
+          Collectors.toMap(Pair::first, blobMetadataByteBufferPair -> SerializationUtils.deserialize(
+              ByteBuffers.toByteArray(blobMetadataByteBufferPair.second()))));
+      return collect.get(blobMetadata.get(0)).get(0);
```

Comment:
should we have a safe check for position indexes?

Reply:
.get(<index>) will through IndexOutofBounds exception. This exception is caught and If the file is corrupted or empty we will return empty ColumnStats Object,

Reply:
We first get the blob metadata:
This is a List that will contain all the blobs. Currently we are writing all stats to only one blob. Therefore we get the blobmetadata at index 0;

Next using this blobmetadata info as key , we get the the array list. Currently its just one element in this list.


![image](https://github.com/apache/hive/assets/26441677/0f69d23e-ac88-4a57-98ac-ee4c48119246)


Map<BlobMetadata, List<ColumnStatistics>>

Reply:
@simhadri-g, why it's not Map<BlobMetadata, ColumnStatistics>? should we serialize List<ColumnStatisticsObj> instead of ColumnStatistics object?

Reply:
Hive provides it as a list of column stats, we are storing this: https://github.com/apache/hive/blame/master/ql/src/java/org/apache/hadoop/hive/ql/stats/ColStatsProcessor.java#L201